src/plugins/intel_gpu/src/graph/network.cpp

```diff
@@ -659,7 +659,9 @@ network::output_chains_map::iterator network::add_output_chain(std::shared_ptr<p
             } else {
                 const auto& mem_dep = dep.first->output_memory();
                 // Add dep inst to the chain when dep's output is not allocated yet.
-                if (!dep.first->outputs_allocated() || eng.is_the_same_buffer(mem_orig, mem_dep)) {
+                if (!dep.first->outputs_allocated()
+                    || (dep.first->outputs_allocated() && !p_inst->outputs_allocated())
```

Comment:
isnt it enough with the second condition only? !p_inst->outputs_allocated()

Reply:
Done